.github/workflows/libcxx-build-and-test.yaml

```diff
@@ -199,6 +201,38 @@ jobs:
             **/CMakeError.log
             **/CMakeOutput.log
             **/crash_diagnostics/*
+
+  macos:
+    runs-on: macos-latest
```

Comment:
I recommend using macos-14 here, so we can control when we update to the next macos version.